src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -36,16 +36,17 @@ class HDF5IO(HDMFIO):
 
     __ns_spec_path = 'namespace'  # path to the namespace dataset within a namespace group
 
-    @docval({'name': 'path', 'type': (str, Path), 'doc': 'the path to the HDF5 file'},
-            {'name': 'manager', 'type': (TypeMap, BuildManager),
-             'doc': 'the BuildManager or a TypeMap to construct a BuildManager to use for I/O', 'default': None},
+    @docval({'name': 'path', 'type': (str, Path), 'doc': 'the path to the HDF5 file', 'default': None},
             {'name': 'mode', 'type': str,
              'doc': ('the mode to open the HDF5 file with, one of ("w", "r", "r+", "a", "w-", "x"). '
                      'See `h5py.File <http://docs.h5py.org/en/latest/high/file.html#opening-creating-files>`_ for '
-                     'more details.')},
+                     'more details.'),
+             'default': 'r'},
+            {'name': 'manager', 'type': (TypeMap, BuildManager),
+             'doc': 'the BuildManager or a TypeMap to construct a BuildManager to use for I/O', 'default': None},
```

Comment:
Moving this argument in the argument list for `HDF5IO` is technically a breaking change, though a very minor one. I agree with making the change because `HDF5IO(path, mode, manager, ...)` follows convention better than `HDF5IO(path, manager, mode)`. But we should probably bump HDMF to 3.4.0 in this case.

Reply:
Actually never mind. I see what happened here. "mode" did not have a default value before, so it would be parsed as `HDF5IO(path, mode, manager, ...)`. Ignore these comments.

Reply:
Actually, I don't think it is a breaking change. At least the intention is that it isn't. This is a quark of docval- args with no default go first in positional arguments even if they aren't listed first in docval. Once I added default values for everything, I needed to move "manager" later in the list so that it *didn't* cause a breaking change

Reply:
I'm glad you brought this up! I should have mentioned this in the PR comments